core/iwasm/aot/aot_runtime.h

```diff
@@ -105,14 +105,25 @@ typedef struct AOTFunctionInstance {
     char *func_name;
     uint32 func_index;
     bool is_import_func;
+
     union {
         struct {
             AOTFuncType *func_type;
-            /* function pointer linked */
             void *func_ptr;
         } func;
+        /* setup by wasm_native */
         AOTImportFunc *func_import;
     } u;
+
+    /* from other .wasm */
+    AOTModuleInstance *import_module_inst;
+    struct AOTFunctionInstance *import_func_inst;
+    /* copy it from AOTImportFunc */
+    bool call_conv_raw;
+    /* write it from wasm_c_api */
+    bool call_conv_wasm_c_api;
+    /* AOTModuleInstance collects it to form c_api_func_imports */
+    CApiFuncImport import_func_c_api;
```

Comment:
These fields are only required by import functions, adding them here will increase memory consumption when non-import function instances are created, could we get them or most of them from the `AOTImportFunc *func_import`?

Reply:
It's quite challenging to do that.

`AOTModuleInstance *import_module_inst` and `struct AOTFunctionInstance *import_func_inst` are used for instantiation linking with other modules, so we need to retain both here.

`bool call_conv_raw` is utilized for registered native symbols and can be taken from `AOTImportFunc`.

`bool call_conv_wasm_c_api` and `CApiFuncImport import_func_c_api` are created by the wasm_c_api APIs. Both should be capable of instantiation linking.

Therefore, only one `bool` field is duplicated from `AOTImportFunc`. The other fields need to remain in `AOTFunctionInstance` to meet the requirements of instantiation linking.